src/autocomplete.js

```diff
@@ -49,11 +50,61 @@ export default class Autocomplete {
     this.input.removeEventListener('blur', this.onInputBlur)
     this.input.removeEventListener('input', this.onInputChange)
     this.results.removeEventListener('mousedown', this.onResultsMouseDown)
+    this.results.removeEventListener('click', this.onResultsClick)
+  }
+
+  sibling(next: boolean): Element {
+    const options = Array.from(this.results.querySelectorAll('[role="option"]'))
+    const selected = this.results.querySelector('[aria-selected="true"]')
+    const index = options.indexOf(selected)
+    const sibling = next ? options[index + 1] : options[index - 1]
+    const def = next ? options[0] : options[options.length - 1]
+    return sibling || def
+  }
+
+  select(target: Element) {
+    for (const el of this.results.querySelectorAll('[aria-selected="true"]')) {
+      el.removeAttribute('aria-selected')
+    }
+    target.setAttribute('aria-selected', 'true')
+    this.input.setAttribute('aria-activedescendant', target.id)
   }
 
   onKeydown(event: KeyboardEvent) {
-    if (event.key === 'Escape') {
-      this.container.open = false
+    switch (event.key) {
+      case 'Escape':
+        this.container.open = false
+        event.preventDefault()
+        break
+      case 'ArrowDown':
+        this.select(this.sibling(true))
+        event.preventDefault()
+        break
+      case 'ArrowUp':
+        this.select(this.sibling(false))
+        event.preventDefault()
+        break
+      case 'n':
+        if (event.ctrlKey) {
+          this.select(this.sibling(true))
+          event.preventDefault()
+        }
+        break
+      case 'p':
```

Comment:
Isn't <kbd>Ctrl-P</kbd> a shortcut for Print on Windows? There is a chance (I'm guessing) that some browsers might not let this keyboard combination being overriden by JS.